angularjs/app/app.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+// Declare app level module which depends on views, and components
+
+var app = angular.module('myApp', [
+  'ngRoute',
+  'myApp.view1',
+  'myApp.view2',
+  'myApp.view3',
+  'myApp.version',
+]);
+
+app.config(['$locationProvider', '$routeProvider', function($locationProvider, $routeProvider) {
+  $locationProvider.hashPrefix('!');
+  $routeProvider.otherwise({redirectTo: '/view1'});
+}]);
+
+// Wootric code
+app.factory('wootricFactory', function () {
+  window.wootricSettings = {
+    email: 'nps@example.com',// TODO: The current logged in user's email address.
+    created_at: 1234567890, // TODO: The current logged in user's sign-up date as a 10 digit Unix timestamp.
+    account_token: 'NPS-xxxxxxx' // This is your unique account token.
+  };
+  var i=new Image;i.src="//d8myem934l1zi.cloudfront.net/pixel.gif?account_token="+window.wootricSettings.account_token+"&email="+encodeURIComponent(window.wootricSettings.email)+"&created_at="+window.wootricSettings.created_at+"&url="+encodeURIComponent(window.location)+"&random="+Math.random()
+
+  return {
+    run: function() {
+      window.wootric_survey_immediately = true; // Shows survey immediately for testing purposes.  TODO: Comment out for production.
+      window.wootric_no_surveyed_cookie = true; // Bypass cookie based throttle for testing purposes.  TODO: Comment out for production.
+      window.wootric("run");
+    }
+  }
+});
+// End Wootric code
+
+app.controller('myController', [
+  'wootricFactory',
+  '$scope',
+  '$location',
+  function(wootricFactory, $scope, $location){
+    $scope.mode = 1;
+    $scope.$on('$routeChangeStart', function(){
+      var path = $location.path()
+      $scope.mode = parseInt(path.slice(-1));
+    });
+
+    console.log($scope);
+    console.log($scope.mode);
```

Comment:
Delete these `console.log` statements and cleanup the code in the controller so it's as simple as possible.